cardano-api/src/Cardano/Api/SpecialByron.hs

```diff
@@ -0,0 +1,185 @@
+-- | Special Byron values that we can submit to a node to propose an update proposal
+-- or to vote on an update proposal. These are not transactions.
+--
+module Cardano.Api.SpecialByron
+  ( ByronUpdateProposal(..),
+    serialiseByronUpdateProposal,
+    deserialiseByronUpdateProposal,
+    updateProposalToGenTx,
+    ParametersToUpdate(..),
+    createProtocolParametersUpdate,
+    ByronVote(..),
+    createByronUpdateProposal,
+    createByronVote,
+    serialiseByronVote,
+    deserialiseByronVote,
+    voteToGenTx,
+  ) where
+
+import           Cardano.Prelude (void)
+import           Prelude
+
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString.Lazy as LB
+import qualified Data.Map.Strict as M
+import           Data.Word
+import           Numeric.Natural
+
+
+import           Cardano.Api.Key
+import           Cardano.Api.KeysByron
+import           Cardano.Api.NetworkId (NetworkId, toByronProtocolMagicId)
+
+import qualified Cardano.Binary as CBOR
+import           Cardano.Chain.Common (LovelacePortion, TxFeePolicy)
+import           Cardano.Chain.Slotting
+import           Cardano.Chain.Update
+import           Cardano.Crypto (SafeSigner, noPassSafeSigner)
+import           Ouroboros.Consensus.Byron.Ledger.Block (ByronBlock)
+import qualified Ouroboros.Consensus.Byron.Ledger.Mempool as Mempool
+
+-- | Byron era update proposal
+
+newtype ByronUpdateProposal a = ByronUpdateProposal { unByronUpdateProposal :: AProposal a}
```

Comment:
I think we can get rid of the type param, and use just `AProposal ByteString`.

In the `createByronUpdateProposal` case it will mean we need to convert from `AProposal ()` to `AProposal ByteString` but I think there's a function in the Byron ledger code to do that.

Doing this for both `ByronUpdateProposal` and `ByronVote` will mean we can make both an instance of `SerialiseAsRawBytes`, which will mean we don't need to export `serialiseByronUpdateProposal` etc.